### PR TITLE
verifier: update to the latest Tendermint light client

### PR DESCRIPTION
**Pull request by melekes**
v0.33.1-dev2

### CODE REVIEW

File: relayer/verifier.go [hunk 1]
```diff
@@ -74,41 +78,25 @@ func (c *Chain) UpdateLiteDBToLatestHeader() error {
...
-			defer wg.Done()
-			err := chain.UpdateLiteDBToLatestHeader()
-			if err != nil {
-				errs.Lock()
-				errs.Map[chain] = err
-				errs.Unlock()
+		chain := chain
+		g.Go(func() error {
+			if err := chain.UpdateLiteDBToLatestHeader(); err != nil {
+				return fmt.Errorf("failed to update chain %s: %w", chain, err)
```
Comment (jackzampolin) [inline relayer/verifier.go:85]: This is a really cool! Haven't seen this `errgroup` thing before. Exactly what I wanted to do here.

File: relayer/verifier.go [hunk 2]
```diff
@@ -219,38 +207,30 @@ func (c *Chain) GetLatestLiteHeader() (*tmclient.Header, error) {
...
+		chain := chain
+		g.Go(func() error {
+			h, err := chain.GetLatestLiteHeader()
 			if err != nil {
-				hs.Errs = append(hs.Errs, err)
+				return fmt.Errorf("failed to get latest header for chain %s: %w", chain, err)
 			}
-			wg.Done()
-		}(hs, &wg, chain)
+			m[chain.ChainID] = h
```
Comment (jackzampolin) [inline relayer/verifier.go]: I think this is going to run into concurrent map access issues right?

Comment (jackzampolin) [reply relayer/verifier.go]: So we still need the mutex here to ensure no issues?

Comment (melekes) [reply relayer/verifier.go]: sync.Map?

**Review by jackzampolin — APPROVED** [review]

File: relayer/verifier.go (hunk 1 above)
Comment (jackzampolin) [reply relayer/verifier.go:85]: Update, this call just hangs now

**Pull request merged**